### ISSUE TITLE
Add option to explicitly set number of dimensions on read/write (specific use case: DCD)

### ISSUE THREAD

**Issue by shannon-moran** [glotzerlab/garnett#135]
## Description

- To make DCD files that dump the orientation into the x-component, typically the traj box needed to be set to 3 dimensions, which is what then gets picked up the garnett read/write (it's possible that this is not a universal problem, but was the case at one point)
- Currently, the only way to get garnett to interpret DCD files as 2D with the orientation in the z-component is to manually input a 2D frame (code snippet below)
- Adding dimensions as an optional keyword to the read/write calls would make this a much simpler option

## Proposed solution

- Adding dimensions as an optional keyword to the read/write calls would make this a much simpler option, e.g.:
```
with garnett.read('dump.dcd', dimensions=2) as traj:
```

## Additional context

Currently, the only way I can get the orientation to properly be read out of the z-component and included in the new `.gsd` file is:
```
# Solution to dcd reader not catching 2D setting
with garnett.read('dump.dcd') as traj:
    config_frame = traj[0]
    config_frame.box.dimensions = 2

dcd_reader=garnett.reader.DCDFileReader()
with open('dump.dcd', 'rb') as dcdfile:
    traj=dcd_reader.read(dcdfile,frame=config_frame)
    for frame in range(len(traj)):
        traj[frame].shapedef = {"A": garnett.shapes.SpheropolygonShape(vertices=vertices,rounding_radius=rounding_radius,color='FFCB05')}
    garnett.write(traj,'dump.gsd')
```

If there were a dimensions option on the read, then one could write the following.
```
with garnett.read('dump.dcd', dimensions=2) as traj:
    for frame in range(len(traj)):
        traj[frame].shapedef = {etc}
    garnett.write(traj,'dump.gsd')
```

I realize this might only be useful for `.dcd` files, but I think having an explicit flag would be useful, and shouldn't be a huge lift to implement.

**Comment by bdice** [comment]
@shannon-moran If you provide a frame (from another file, e.g. an XML or GSD frame) with a 2D box, the [DCD reader class will detect that](https://garnett.readthedocs.io/en/stable/readerswriters.html#garnett.reader.DCDFileReader) and interpret the z-component as an orientation.

If you wanted to use the auto-reader `garnett.read`, this feature is already implemented for GSD template frames but not enabled for DCD frames. I will create a tiny pull request that you can test out.

**Comment by shannon-moran** [comment]
@bdice right, I do use the DCD reader class in this example, but think it would be great if at least the frame option were available for DCD in the `garnett.read` auto-reader like you've done in #136. I'll take a look.
